2-WebApp-graph-user/2-5-HybridFlow/wwwroot/js/RedeemCode.js

```diff
@@ -0,0 +1,27 @@
+// IIFE to acquire token from auth code sent through server.
+(function () {
+    const code = document.cookie.split(';').reduce((cookies, cookie) => {
+        const [name, value] = cookie.split('=').map(c => c.trim());
+        cookies[name] = value;
+        return cookies;
+    }, {})['Microsoft.Identity.Hybrid.Authentication.Cookie'];
+
+    const scopes = ["user.read"];
+
+    if (!!code) {
+        msalInstance.acquireTokenByCode({
+            code,
+            scopes
+        });
```

Comment:
A couple things here:

1. The application should invoke and await `handleRedirectPromise` before doing this (so that way redirect based interaction is processed correctly, to prevent a redirect loop).
2. While it should be rare, you should show how to handle errors and invoke an interaction method.

You can see these here (ignore the ssoSilent logic): https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/samples/msal-browser-samples/HybridSample/views/client-redirect.hbs#L16-L63